File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set(begin(nums), end(nums));
        int max = 0;
        int current_seq = 1;
        for(int num : set) {
            int prev = num - 1;
            if(set.find(prev) == set.end()) { //Start of set
                int current = num + 1;
                while(set.find(current) != set.end()) {
                    cout << current << endl;
                    current += 1;
                    current_seq += 1;
                }
            if(current_seq > max) {
                max = current_seq;
            }
            current_seq = 1;
            }
        }
        return max;

    }
};

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        result = [0] * len(nums)
        result[0] = nums[0]
        result[1] = max(nums[0], nums[1])
        for i in range(2, len(nums)):
            current = nums[i] + result[i - 2]
            result[i] = max(current, result[i - 1])
        return result[-1]